### PR TITLE
Am 129 password history jest testing

### PR DESCRIPTION
**Pull request by farmborough**
## :id: 
Fixes [https://gravitee.atlassian.net/browse/AM-129](AM-129)

## :pencil2: 
Tests for both the Management and Gateway reset password endpoints. Feel free to ignore the generate api `ts` files; the tests to review are `reset-password.jest.spec` and `password-history.jest.spec`.

### CODE REVIEW

File: gravitee-am-test/specs/management/password-history.jest.spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+});
+
+describe("Testing password history...", () => {
+    describe("when password history is enabled", () => {
+
+        beforeAll(async () => {
+            domain = await patchDomain(domain.id, accessToken, {
+                "passwordSettings": {
+                    "passwordHistoryEnabled": true,
+                    "oldPasswords": 3
```
Comment (leleueri) [inline gravitee-am-test/specs/management/password-history.jest.spec.ts:47]: In order to fully validate the feature, maybe we should also test that after 3 reset password the first one can be reused? wdyt?

File: gravitee-am-test/specs/gateway/reset-password.jest.spec.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+beforeAll(async () => {
+    const adminTokenResponse = await requestAdminAccessToken();
+    accessToken = adminTokenResponse.body.access_token;
+    domain = await createDomain(accessToken, "test-reset-password", "test Gateway reset password with password history");
+});
+
+describe("Gateway reset password", () => {
+    beforeAll(async () => {
+        domain = await patchDomain(domain.id, accessToken, {
```
Comment (leleueri) [inline gravitee-am-test/specs/gateway/reset-password.jest.spec.ts:50]: PasswordPolicy can be overided at application level.
In this simulation we only check that the domain settings is working but IMHO we should also have a dedicated test to validate the usage of the application settings.

In both case we should check that the first password can be reused after 3 resetPassword.

**Review by leleueri — CHANGES_REQUESTED** [review]

File: gravitee-am-test/specs/management/password-history.jest.spec.ts (hunk 1 above)
Comment (farmborough) [reply gravitee-am-test/specs/management/password-history.jest.spec.ts:47]: Agree. but there isn't really time at this stage to add anything else to the test (given that I'm on support and the only person fully in the team today).

**Review by leleueri — APPROVED** [review]
Can be merge if test cases identified during the review are managed by the task "JEST: forgot password"

File: gravitee-am-test/specs/management/password-history.jest.spec.ts (hunk 1 above)
Comment (farmborough) [reply gravitee-am-test/specs/management/password-history.jest.spec.ts:53]: Done.

File: gravitee-am-test/specs/gateway/reset-password.jest.spec.ts (hunk 2 above)
Comment (farmborough) [reply gravitee-am-test/specs/gateway/reset-password.jest.spec.ts:56]: Done. Application stuff can be done during the quality sprint.

**Pull request merged**